src/main/java/org/broadinstitute/hellbender/tools/LocalAssembler.java

```diff
@@ -0,0 +1,2306 @@
+        for ( final Traversal down : downExtensions ) {
+            for ( final Traversal up : upExtensions ) {
+                if ( scaffolds.size() >= TOO_MANY_SCAFFOLDS ) {
+                    throw new AssemblyTooComplexException();
+                }
+                scaffolds.add(Traversal.combineOverlappers(up.rc(), down, traversal.getContigs().size()));
+            }
+        }
+    }
+
+    private static void walkTraversals( final Traversal traversal,
+                                        final boolean[] touched,
+                                        final Set<Contig> startingContigSet,
+                                        final Map<Contig, List<Integer>> traversalsByFirstContig,
+                                        final List<Traversal> allTraversals,
+                                        final List<Traversal> extensions ) {
+        final Contig firstContig = traversal.getFirstContig();
+        final List<Integer> indexList;
+        if ( startingContigSet.contains(firstContig) ||
+                traversal.isInextensible() ||
+                (indexList = traversalsByFirstContig.get(traversal.getLastContig())) == null ) {
+            extensions.add(traversal);
+            return;
+        }
+        startingContigSet.add(firstContig);
+        for ( int idx : indexList ) {
+            final Traversal extension;
+            if ( idx >= 0 ) {
+                extension = allTraversals.get(idx);
+                touched[idx] = true;
+            } else {
+                final int rcIdx = ~idx;
+                extension = allTraversals.get(rcIdx).rc();
+                touched[rcIdx] = true;
+            }
+            walkTraversals(Traversal.combine(traversal, extension), touched, startingContigSet,
+                            traversalsByFirstContig, allTraversals, extensions );
+        }
+        startingContigSet.remove(firstContig);
+    }
+
+    private static void removeTriviallyDifferentTraversals(
+                                            final Collection<Traversal> allTraversals ) {
+        if ( allTraversals.isEmpty() ) {
+            return;
+        }
+        final TreeSet<Traversal> sortedTraversals = new TreeSet<>(new TraversalEndpointComparator());
+        for ( final Traversal traversal : allTraversals ) {
+            sortedTraversals.add(traversal);
+            sortedTraversals.add(traversal.rc());
+        }
+        final Iterator<Traversal> traversalIterator = sortedTraversals.iterator();
+        Traversal prevTraversal = traversalIterator.next();
+        while ( traversalIterator.hasNext() ) {
+            final Traversal curTraversal = traversalIterator.next();
+            if ( isTriviallyDifferent(prevTraversal, curTraversal) ) {
+                traversalIterator.remove();
+            } else {
+                prevTraversal = curTraversal;
+            }
+        }
+        sortedTraversals.removeIf(Traversal::isRC);
```

Comment:
We added both strands at the beginning, and I thought I had the sort order arranged so that, for any given pair that was trivially different, at least one of the +strand guys survived removal.  Alas, that wasn't true. It's rewritten to avoid that assumption.